### PR TITLE
👹 Havoc: Refactor channels tests to embrace chaos

### PR DESCRIPTION
**Pull request by madmax983**
🌪️ **The Trigger:**
The `test_channels_zero_capacity_regression` and `test_channels_capacity_fuzzing` tests for the `Channels` module were testing the happy path (checking for no panics) instead of embracing edge-cases and chaos scenarios.

📉 **The Stack Trace:**
N/A - This was a test health/design issue, not a runtime crash. The existing tests failed to adequately stress the `Channels` implementation, particularly regarding concurrency and bounded capacity behavior under extreme load.

🧪 **Reproduction:**
By replacing the simple assertion loops with a high-concurrency setup (10 tasks sending 100 messages each, 10 tasks reading) on a 0-capacity channel, we actively create and verify the expected `Lagged` errors. We also explicitly test the edge-case of sending messages to a channel with no subscribers expecting an error rather than just avoiding a panic.

😈 **Comment:**
This ensures our channels hold up under heavy contention and properly bubble up errors when consumers inevitably fall behind or when messages are fired into the void. Chaos guarantees stability!

---
*PR created automatically by Jules for task [16665335570032543125](https://jules.google.com/task/16665335570032543125) started by @madmax983*

### CODE REVIEW

File: autumn/tests/chaos_channels_proptest.rs [hunk 1]
```diff
@@ -10,23 +11,50 @@ proptest! {
 
         // This should never panic on any capacity (see explicit zero test)
         let tx = channels.sender("test_channel");
-        let _rx = channels.subscribe("test_channel");
-        prop_assert!(tx.send("test").is_ok());
+
+        // Edge case: Sending with no subscribers should cleanly error, not panic
+        let res = tx.send("test");
+        prop_assert!(res.is_err());
```
[the remark this answers is not in the document]
Comment (madmax983) [reply autumn/tests/chaos_channels_proptest.rs:17]: > **P2 Badge Exercise capacity behavior in the fuzz test**
> 
> `test_channels_capacity_fuzzing` now sends on a channel with no subscribers and only asserts `is_err()`, which does not depend on `capacity`. This makes the proptest effectively vacuous for buffer-capacity regressions: failures in capacity-dependent send/receive behavior (the thing this test name implies) can slip through while this test still passes for all generated values. Keep at least one subscriber in this property test (or split out a separate no-subscriber test) so generated capacities actually affect what is being validated.
> 
> Useful? React with 👍 / 👎.

@jules
